packages/flutter_tools/lib/src/flutter_project_metadata.dart

```diff
@@ -51,21 +51,18 @@ FlutterProjectType? stringToProjectType(String value) {
 }
 
   /// Verifies the expected yaml keys are present in the file.
-  bool _validateMetadataMap(Object? yamlRoot, Map<String, Type> validations, Logger logger) {
-    if (yamlRoot != null && yamlRoot is! YamlMap) {
-      return false;
-    }
-    final YamlMap map = yamlRoot! as YamlMap;
+  bool _validateMetadataMap(YamlMap map, Map<String, Type> validations, Logger logger) {
     bool isValid = true;
     for (final MapEntry<String, Object> entry in validations.entries) {
       if (!map.keys.contains(entry.key)) {
         isValid = false;
         logger.printTrace('The key `${entry.key}` was not found');
         break;
       }
-      if (map[entry.key] != null && (map[entry.key] as Object).runtimeType != entry.value) {
+      final Object? metadataValue = map[entry.key];
```

Comment:
This was the bug: when `map[entry.key]` was null, this method was returning true (valid).
Instead, check the `map[entry.key]` `runtimeType` even when it is null so it is marked as invalid and prints `The value of key "x" in .metadata was expected to be y but was Null`